File: LeetCode/Java Solutions/Maximal Score After Applying K Operations/maximal.java
import java.util.PriorityQueue;

class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>((a, b) -> b - a);
        
        // Add all elements to the max heap
        for (int num : nums) {
            maxHeap.offer(num);
        }
        
        long score = 0;
        
        // Perform k operations
        for (int i = 0; i < k; i++) {
            int max = maxHeap.poll();
            score += max;
            maxHeap.offer((max + 2) / 3); // Equivalent to ceil(max / 3)
        }
        
        return score;
    }
}

File: LeetCode/Python Solutions/Maximal Score After Applying K Operations/maximal.py
import heapq

class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        # Convert nums to a max heap
        max_heap = [-num for num in nums]
        heapq.heapify(max_heap)
        
        score = 0
        
        # Perform k operations
        for _ in range(k):
            max_val = -heapq.heappop(max_heap)
            score += max_val
            heapq.heappush(max_heap, -((max_val + 2) // 3)) # Equivalent to ceil(max_val / 3)
        
        return score